app/models/hackbot/interactions/text_conversation.rb

```diff
@@ -34,8 +34,27 @@ def file_to_channel(filename, file)
         send_file(data['channel'], filename, file)
       end
 
+      def slack_id
+        if event
+          event[:user]
+        elsif data['channel']
+          # This method can accurately find the Slack ID of a user if the
+          # conversation is a PM. HOWEVER, it will not work in any other type
+          # context.
+          dm_channel = channels.find { |im| im[:id] == data['channel'] }
+          dm_channel ? dm_channel[:user] : nil
+        end
+      end
+
       private
 
+      def channels
+        key = "im.list##{team.bot_access_token}"
```

Comment:
Could you find something else to use as the key here (as talked about in-person.